tmt/base.py

```diff
@@ -1587,13 +1587,28 @@ def import_plan(self) -> Optional['Plan']:
                 else:
                     tmt.utils.git_clone(plan_id.url, destination, self)
                 if plan_id.ref:
-                    self.run(Command('git', 'checkout', plan_id.ref), cwd=destination)
+                    # attempt to evaluate dynamic reference
+                    try:
+                        ref = resolve_dynamic_ref(
+                            workdir=destination,
+                            ref=plan_id.ref,
+                            plan=self,
+                            logger=self._logger
+                            )
+                    except tmt.utils.FileError as error:
+                        raise tmt.utils.DiscoverError(str(error))
+                    if ref:
+                        self.run(Command('git', 'checkout', ref), cwd=destination)
                 if plan_id.path:
                     destination = os.path.join(destination, plan_id.path.lstrip("/"))
                 node = fmf.Tree(destination).find(plan_id.name)
 
             # Use fmf cache for exploring plans (the whole git repo is not needed)
             else:
+                # fmf does not support dynamic ref so we need to lie
```

Comment:
I don't like this but don't know how to handle it given that fmf does not support dynamic ref.

Reply:
If it stays like this, I'd appreciate a more extended comment, it took me a while to realize why `ref` is thrown away & `_ref_expanded`, constructed above, is not used at all.

Reply:
@happz I have added more comments and a warning, could you please take a look?

Reply:
It's much better, thanks.